webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/RestUtils.scala

```diff
@@ -145,12 +145,9 @@ object RestUtils extends Loggable {
                   ( "id"     -> id ) ~
                   ( "result" -> status.status ) ~
                   ( status.container   ->  message )
-    val content : JsExp = new JsExp {
-      lazy val toJsCmd = if(prettify) render(json) else compactRender(json)
-    }
-
-    JsonResponse(content,List(),List(), status.code)
+    val content = if(prettify) render(json) else compactRender(json)
 
+    InMemoryResponse(content.getBytes(),List(),List(), status.code)
```

Comment:
Why changing the response type ? I think it needs header stating its a json response

Reply:
JsonResponse is defining two headers: 


    InMemoryResponse(bytes, ("Content-Length", bytes.length.toString) :: ("Content-Type", "application/json; charset=utf-8") :: headers, cookies, code)

You should at least define them

Reply:
the answer was not really a json response: it was text but in a `JsExp` construct. We did that to allow formatting without having to change lift JsonResponse. 
This was breaking test that try to test the actual Json structure, since it was not a json structure but just formatted string. 
I will look for a better solution or if not, include the provided headers.

Reply:
I created a new `JsonResponse` derivative to handle the case